test/unifex/integration_test.exs

```diff
@@ -2,33 +2,57 @@ defmodule Unifex.IntegrationTest do
   use ExUnit.Case, async: true
 
   test "NIF test project" do
+    generate_cpp_code("nif")
+
     test_project("nif", :nif, :c)
     test_project("nif", :nif, :cpp)
+
+    delete_cpp_code("nif")
```

Comment:
Let's rename `test_project` to something like `do_test_project` and move this to a new `test_project` function. Also, delete_cpp_code should be run with `on_exit` https://hexdocs.pm/ex_unit/ExUnit.Callbacks.html#on_exit/2

Reply:
Problem is, that `on_exit` should be called from `setup` function, which is shared between many test functions and each test generates and deletes files from different directory

Reply:
`on_exit` can be called from within a test as well. If you don't use it, the files won't be deleted when the test fails

Reply:
bump